metasv/spades_contig.py

```diff
@@ -8,13 +8,13 @@ def __init__(self, name="", sequence=None):
         self.parse_name(name)
 
     def parse_name(self, name):
-        name_fields = name.strip().split("_")
+        name_match = re.search(r'^(.+)_(\d+)_(\d+)_(INS|DEL)_\d+_NODE_\d+_length_(\d+)_cov_(\d+)',name)
```

Comment:
Thanks for the fix. Could you please change the regex to `r'^(.+)_(\d+)_(\d+)_(INS|DEL)_\d+_NODE_\d+_length_(\d+)_cov_(\d*\.\d+|\d+)'` since that takes care of covs being a floating-point number? Also, it would be better to have the regex pre-compiled outside the class as `pattern = re.compile(r'^(.+)_(\d+)_(\d+)_(INS|DEL)_\d+_NODE_\d+_length_(\d+)_cov_(\d*\.\d+|\d+)'` and then simply using `name_match = pattern.search(name)` to avoid compilation of the pattern everytime.